geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/utils/GeoJsonInference.scala

```diff
@@ -0,0 +1,292 @@
+      parser.nextToken()
+      numRead+=1
+    }
+    is.close()
+
+    // add geometry
+    val mergedGeomType = mergeGeomTypes(geomTypes)
+    val allAttrs = attributes.values.toArray :+ buildGeomAttr(mergedGeomType)
+
+    sftBuilder.addAll(allAttrs)
+    sftBuilder.setName(sftName)
+    sftBuilder.setDefaultGeometry("geometry")
+    sftBuilder.buildFeatureType()
+  }
+
+  def inferConfig(sft: SimpleFeatureType, idFieldName: Option[String] = None): Config = {
+    val attributes = sft.getAttributeDescriptors.map { ad =>
+      val config = AttributeSpec(sft, ad).toConfigMap
+      val attrMap: java.util.HashMap[String,String] = new java.util.HashMap[String,String](config.asJava)
+      val typeVal = attrMap.get("type").toLowerCase()
+      ad.getType.getBinding match {
+        case t if classOf[java.util.List[_]].isAssignableFrom(t) =>
+          attrMap.put("json-type", "list")
+          attrMap.put("transform", "jsonList('string', $0)")
+          attrMap.put("path", "$.properties." + attrMap.get("name"))
+        case g if classOf[Geometry].isAssignableFrom(g) =>
+          val geomType = attrMap.get("type").toLowerCase()
+          attrMap.put("json-type", "geometry")
+          attrMap.put("transform", s"$geomType($$0)")
+          attrMap.put("path", "$.geometry")
+        case _ =>
+          attrMap.put("json-type", typeVal)
+          attrMap.put("path", "$.properties." + attrMap.get("name"))
+      }
+
+      attrMap.put("name", attrMap.get("name"))
+      attrMap
+    }
+
+    baseConfig
+      .withValue(TypeNamePath, ConfigValueFactory.fromAnyRef(sft.getTypeName))
+      .withValue(FieldsPath, ConfigValueFactory.fromIterable(attributes))
+  }
+
+  def mergeMaps(attrsA: Map[String, AttributeDescriptor],
+                attrsB: Map[String, AttributeDescriptor]): Map[String, AttributeDescriptor] = {
+    val (overlappingKeys, disjointKeys) = attrsB.keys.partition(attrsA.contains)
+
+    // start with fields unique to A
+    var newMap = attrsA.filter( p => !attrsB.contains(p._1)).toSeq
+    // Add fields unique to B
+    disjointKeys.foreach{ name => newMap = newMap :+ (name, attrsB(name)) }
+    // Merge overlaps
+    val merged = overlappingKeys.map { name =>
+      val bindingA = attrsA(name).getType.getBinding
+      val bindingB = attrsB(name).getType.getBinding
+      if (bindingB == classOf[jl.Object] || bindingA == bindingB) {
+        (name, attrsA(name))
+      } else if (bindingA == classOf[jl.Object]) {
+        (name, attrsB(name))
+      } else {
+        val JDouble = classOf[jl.Double]
+        val JInteger = classOf[jl.Integer]
+        val JLong = classOf[jl.Long]
+        (bindingA, bindingB) match {
+          case (JDouble, JInteger) => (name, attrsA(name))
+          case (JInteger, JDouble) => (name, attrsB(name))
+          case (JLong, JInteger) => (name, attrsA(name))
+          case (JInteger, JLong) => (name, attrsB(name))
+          case _ => (name, attrsA(name))
```

Comment:
do we have any sense of weighted types? e.g. if a field parsed as a date 9 times but as a string the 10th, would we consider that a date?

Reply:
No notion of weighted types at the moment, though that would be nice to have. There's just a hierarchy of precedence where "empty" is at the bottom. I'll play around with how fun/challenging it would be to add that